File: app/src/main/java/com/application/arenda/MainWorkspace/Activities/ActivityViewImages.java
package com.application.arenda.MainWorkspace.Activities;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import com.application.arenda.Entities.Announcements.ViewAnnouncement.AdapterPagerBigImageView;
import com.application.arenda.R;
import com.github.piasy.biv.BigImageViewer;
import com.github.piasy.biv.loader.glide.GlideImageLoader;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ActivityViewImages extends AppCompatActivity {

    @Nullable
    @BindView(R.id.itemBtnBack)
    ImageButton itemBtnBack;

    @Nullable
    @BindView(R.id.pagerBigImageView)
    ViewPager pagerBigImageView;

    @Nullable
    @BindView(R.id.pagerIndicator)
    TextView pagerIndicator;

    private Uri selectedUri;
    private List<Uri> uriList = new ArrayList<>();

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        BigImageViewer.initialize(GlideImageLoader.with(this));

        setContentView(R.layout.activity_view_image);

        ButterKnife.bind(this);

        Intent intent = getIntent();

        uriList = intent.getParcelableArrayListExtra("CollectionUri");
        selectedUri = intent.getParcelableExtra("SelectedUri");

        pagerBigImageView.setAdapter(new AdapterPagerBigImageView(uriList));

        pagerBigImageView.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                setIndicatorPosition(++position);
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });

        int positionSelectedUri = uriList.indexOf(selectedUri);

        setIndicatorPosition(positionSelectedUri);

        pagerBigImageView.setCurrentItem(positionSelectedUri);

        itemBtnBack.setOnClickListener(v -> onBackPressed());
    }

    @SuppressLint("SetTextI18n")
    private void setIndicatorPosition(int position) {
        pagerIndicator.setText(position + "/" + uriList.size());
    }
}

File: app/src/main/java/com/application/arenda/mainWorkspace/activities/ActivityViewImages.kt
package com.application.arenda.mainWorkspace.activities

import android.annotation.SuppressLint
import android.net.Uri
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.application.arenda.databinding.ActivityViewImageBinding
import com.application.arenda.entities.announcements.viewAnnouncement.AdapterPagerBigImageView
import com.github.piasy.biv.BigImageViewer
import com.github.piasy.biv.loader.glide.GlideImageLoader
import java.util.*

class ActivityViewImages : AppCompatActivity() {
    private var bind: ActivityViewImageBinding? = null

    private var selectedUri: Uri? = null
    private var uriList: List<Uri?> = ArrayList()
    private var adapterPagerBigImageView: AdapterPagerBigImageView? = null

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        bind = ActivityViewImageBinding.inflate(layoutInflater)
        setContentView(bind!!.root)

        BigImageViewer.initialize(GlideImageLoader.with(this))

        initComponents()
        initListeners()
        initAdapter()
    }

    private fun initComponents() {
        val intent = intent
        uriList = intent.getParcelableArrayListExtra("CollectionUri")
        selectedUri = intent.getParcelableExtra("SelectedUri")
        adapterPagerBigImageView = AdapterPagerBigImageView(uriList)
    }

    private fun initListeners() {
        adapterPagerBigImageView!!.setImageClickListener {
            if (bind!!.viewImagePanel.visibility == View.INVISIBLE) {
                bind!!.viewImagePanel.visibility = View.VISIBLE
            } else {
                bind!!.viewImagePanel.visibility = View.INVISIBLE
            }
        }
        bind!!.pagerBigImageView.addOnPageChangeListener(object : OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
            override fun onPageSelected(position: Int) {
                var pos = position
                setIndicatorPosition(++pos)
            }

            override fun onPageScrollStateChanged(state: Int) {}
        })
        bind!!.itemBtnBack.setOnClickListener { onBackPressed() }
    }

    private fun initAdapter() {
        bind!!.pagerBigImageView.adapter = adapterPagerBigImageView
        val positionSelectedUri = uriList.indexOf(selectedUri)
        setIndicatorPosition(positionSelectedUri + 1)
        bind!!.pagerBigImageView.currentItem = positionSelectedUri
    }

    @SuppressLint("SetTextI18n")
    private fun setIndicatorPosition(position: Int) {
        bind!!.pagerIndicator.text = position.toString() + "/" + uriList.size
    }
}